### ISSUE TITLE
[Issue] - Health Checker - you cannot call a method on an expression that has a null - Get-NicInformation

### ISSUE THREAD

**Issue by dpaulson45** [microsoft/CSS-Exchange#1648]
**Describe the issue**
Script is failing out due to the following error message:

```
 It is not possible to call a method on an expression that has a NULL.
Inner Exception: for CallSite.Target(Closure , CallSite , Object , String )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction'3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
```

**Expected behavior**
We should handle the exception.

**Script Output**

```
Get-RemoteRegistryValue Return Value: '24'
Failed to find correct adapterConfiguration for this networkConfig.
GUID: {164EF65D-2AF8-4C4D-A0B4-D260AC23B653} | InterfaceGuid: 
Failed to Health Checker against ServerName
Failed Server List: ServerName
 
 
There appears to have been some errors in the script. To assist with debugging of the script, please send the HealthChecker-Debug_*.txt, HealthChecker-Errors.json, and .xml file to ExToolsFeedback@microsoft.com.


Errors that occurred that wasn't handled
Error Index: 0
 : Es ist nicht möglich, eine Methode für einen Ausdruck aufzurufen, der den NULL hat.
Inner Exception:    bei CallSite.Target(Closure , CallSite , Object , String )
   bei System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   bei System.Management.Automation.Interpreter.DynamicInstruction`3.Run(InterpretedFrame frame)
   bei System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
Position Message: In C:\HealthChecker.ps1:11809 Zeichen:90
+ ... Where-Object { $_.Contains(":") }).Count -ge 1
+                    ~~~~~~~~~~~~~~~~
Script Stack: bei <ScriptBlock>, C:\HealthChecker.ps1: Zeile 11809
bei Get-NicInformation<Process>, C:\HealthChecker.ps1: Zeile 11809
bei Get-AllNicInformation<Process>, C:\HealthChecker.ps1: Zeile 11919
bei Get-NetworkingInformation<Process>, C:\HealthChecker.ps1: Zeile 12094
bei Get-OperatingSystemInformation<Process>, C:\HealthChecker.ps1: Zeile 12434
bei Get-HealthCheckerExchangeServer<Process>, C:\HealthChecker.ps1: Zeile 12479
bei Get-HealthCheckerData, C:\HealthChecker.ps1: Zeile 13208
bei Invoke-HealthCheckerMainReport, C:\HealthChecker.ps1: Zeile 13284
bei <ScriptBlock><End>, C:\HealthChecker.ps1: Zeile 14017
bei <ScriptBlock>, <Keine Datei>: Zeile 1
-----------------------------------




Errors that were handled
Error Index: 0
 : Die Benennung "Get-NetIPConfiguration" wurde nicht als Name eines Cmdlet, einer Funktion, einer Skriptdatei oder eines ausführbaren Programms erkannt. Überprüfen Sie die Schreibweise des Namens, oder ob der Pfad korrekt ist (sofern enthalten), und wiederholen Sie den Vorgang.
Inner Exception:    bei System.Management.Automation.CommandDiscovery.LookupCommandInfo(String commandName, CommandTypes commandTypes, SearchResolutionOptions searchResolutionOptions, CommandOrigin commandOrigin, ExecutionContext context)
   bei System.Management.Automation.CommandDiscovery.LookupCommandProcessor(String commandName, CommandOrigin commandOrigin, Nullable`1 useLocalScope)
   bei System.Management.Automation.ExecutionContext.CreateCommand(String command, Boolean dotSource)
   bei System.Management.Automation.PipelineOps.AddCommand(PipelineProcessor pipe, CommandParameterInternal[] commandElements, CommandBaseAst commandBaseAst, CommandRedirection[] redirections, ExecutionContext context)
   bei System.Management.Automation.PipelineOps.InvokePipeline(Object input, Boolean ignoreInput, CommandParameterInternal[][] pipeElements, CommandBaseAst[] pipeElementAsts, CommandRedirection[][] commandRedirections, FunctionContext funcContext)
   bei System.Management.Automation.Interpreter.ActionCallInstruction`6.Run(InterpretedFrame frame)
   bei System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   bei System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
Position Message: In C:\HealthChecker.ps1:11651 Zeichen:47
+                     $networkIpConfiguration = Get-NetIPConfiguration @params | W ...
+                                               ~~~~~~~~~~~~~~~~~~~~~~
Script Stack: bei Get-NetworkConfiguration<Process>, C:\HealthChecker.ps1: Zeile 11651
bei Get-AllNicInformation<Process>, C:\HealthChecker.ps1: Zeile 11880
bei Get-NetworkingInformation<Process>, C:\HealthChecker.ps1: Zeile 12094
bei Get-OperatingSystemInformation<Process>, C:\HealthChecker.ps1: Zeile 12434
bei Get-HealthCheckerExchangeServer<Process>, C:\HealthChecker.ps1: Zeile 12479
bei Get-HealthCheckerData, C:\HealthChecker.ps1: Zeile 13208
bei Invoke-HealthCheckerMainReport, C:\HealthChecker.ps1: Zeile 13284
bei <ScriptBlock><End>, C:\HealthChecker.ps1: Zeile 14017
bei <ScriptBlock>, <Keine Datei>: Zeile 1
-----------------------------------

```

**Additional context**

Need to review this code and understand why we don't `continue` after the `$null` check.
```
                        if ($null -eq $adapterConfiguration) {
                            Write-Verbose "Failed to find correct adapterConfiguration for this networkConfig."
                            Write-Verbose "GUID: $($networkConfig.GUID) | InterfaceGuid: $($networkConfig.InterfaceGuid)"
                        }

                        $ipv6Enabled = ($adapterConfiguration.IPAddress | Where-Object { $_.Contains(":") }).Count -ge 1
```

**Comment by dpaulson45** [comment]
This appears to have been resolved. This issue was created after #870 and should have been a duplicate of it by the looks of it. This issue appears to have been fixed by placing the required code in a `else` statement to avoid issues, but yet still created the object to be added to the list. 

See PR #1716
